Plugins/AWSLambdaPackager/PluginUtils.swift

```diff
@@ -0,0 +1,145 @@
+// ===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// ===----------------------------------------------------------------------===//
+
+import Dispatch
+import PackagePlugin
+
+#if canImport(FoundationEssentials)
+import FoundationEssentials
+#else
+import struct Foundation.URL
+import struct Foundation.Data
+import struct Foundation.CharacterSet
+import class Foundation.Process
+import class Foundation.Pipe
+#endif
+
+struct Utils {
+    @discardableResult
+    static func execute(
+        executable: URL,
+        arguments: [String],
+        customWorkingDirectory: URL? = nil,
+        logLevel: ProcessLogLevel
+    ) throws -> String {
+        if logLevel >= .debug {
+            print("\(executable.absoluteString) \(arguments.joined(separator: " "))")
+        }
+
+        // this shared global variable is safe because we're mutating it in a dispatch group
+        // https://developer.apple.com/documentation/foundation/process/1408746-terminationhandler
```

Comment:
I think the comment here is misleading. The important thing here is the following. We write to the variable from a single SerialDispatchQueue here. We wait until the process is run below `process.waitUntilExit()`. This means no further writes to `output` will happen. This makes it save for us to read the output.

Reply:
We should switch to the new Foundation Process API as soon as it is available:
https://github.com/apple/swift-foundation/blob/main/Proposals/0007-swift-subprocess.md

Reply:
Changed to 
```
        // We need to use an unsafe transfer here to get the fd into our Sendable closure.
        // This transfer is fine, because we write to the variable from a single SerialDispatchQueue here.
        // We wait until the process is run below process.waitUntilExit().
        // This means no further writes to output will happen.
        // This makes it save for us to read the output
```

Reply:
Created https://github.com/swift-server/swift-aws-lambda-runtime/issues/364 to keep track of `SubProcess`